build-version.cmake

```diff
@@ -28,7 +28,7 @@ if (EXISTS ${LOMSE_ROOT_DIR}/.git)
       WORKING_DIRECTORY "${LOMSE_ROOT_DIR}"
       OUTPUT_VARIABLE LOMSE_VERSION_BUILD
       OUTPUT_STRIP_TRAILING_WHITESPACE)
-    execute_process(COMMAND "${GIT_EXECUTABLE}" describe --tags --long --dirty
+    execute_process(COMMAND "${GIT_EXECUTABLE}" describe --tags --long --dirty --all
```

Comment:
Not sure if this is correct. I have to test and investigate. On my local branch, when I execute `git describe --tags --long --dirty` I get `0.27.0-114-g965c2f91` but when I execute  `git describe --tags --long --dirty --all` the result is `heads/changes-0-g965c2f91` and the version is lost! Why did you need to change this?

Reply:
It was an attempt to work around the fatal error as shown below...
```
Grants-iMac:lomse grant$ git describe --tags --long --dirty
fatal: No names found, cannot describe anything.
Grants-iMac:lomse grant$ git describe --tags --long --dirty --all
heads/master-0-gc576c38
```

Reply:
I've reverted this change for now

Reply:
Thank you. Now I understand the problem. I assumed that the repo always contains the tags and thus, that it was feasible to extract the version from the latest tag. But as this link explains (https://stackoverflow.com/questions/4916492/git-describe-fails-with-fatal-no-names-found-cannot-describe-anything) current solution doesn't work for shallow clones.

By adding `--all` you avoid the error for shallow clones, but prevents also that it works for clones with tags, making that code useless. So adding `--all` is not the solution. But it is necessary to avoid the 'fatal: No names found,..." message as not only it is not fatal but also it is irrelevant! I will investigate how to do it. So for now better leave it without the `--all` as you already did. Thank you.